gradio/blocks.py

```diff
@@ -2241,3 +2246,29 @@ def queue_enabled_for_fn(self, fn_index: int):
         if self.dependencies[fn_index]["queue"] is None:
             return self.enable_queue
         return self.dependencies[fn_index]["queue"]
+
+
+def is_update():
+    from gradio import context
+
+    return hasattr(context.thread_data, "blocks")
+
+
+def updateable(fn):
```

Comment:
Maybe we can automatically wrap the init with updateable in a metaclass so that custom component authors don't even have to worry about this when creating their own components.

Reply:
was trying to think how to do this. Not familiar with metaclasses, will take a look!